lib/plugins/package/lib/zipService.js

```diff
@@ -142,47 +150,24 @@ function excludeNodeDevDependencies(servicePath) {
       return !isNodeModulesDir;
     });
 
-    return BbPromise.map(packageJsonPaths, (packageJsonPath) => {
+    // NOTE: using mapSeries here for a sequential computation (w/o race conditions)
+    return BbPromise.mapSeries(packageJsonPaths, (packageJsonPath) => {
       // the path where the package.json file lives
       const fullPath = path.join(servicePath, packageJsonPath);
       const dirWithPackageJson = fullPath.replace(path.join(path.sep, 'package.json'), '');
 
-      const childProc = childProcess.spawn('npm',
-        ['ls', '--dev=true', '--parseable=true', '--silent'],
+      return childProcess.execAsync(
+        `npm ls --dev=true --parseable=true --silent >> ${nodeDevDepFile}`,
```

Comment:
The random number you're using does certainly help though. Didn't notice this before.